### ISSUE TITLE
Refactor to improve utilities to get source location indices.

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#8097]
See :
- https://github.com/stylelint/stylelint/pull/8066
- https://github.com/stylelint/stylelint/pull/8066#discussion_r1821578388

This change adds a couple of new utilities that get the source location indicies of various parts of AST nodes.

Having these as shared utilities has several advantages:
- shared code
- unit tests

But the utilities themselves are often quite small.
Sometimes as simple as this:

```js
/**
 * @param {import('postcss').AtRule} atRule
 * @returns {number}
 */
export default function atRuleRawsAfterNameIndex(atRule) {
	// Initial 1 is for the `@`
	return 1 + atRule.name.length;
}
```

This code is clearly trivial, but knowing that you need this code is not trivial.

I really like having these utilities.
But I can understand that the number of files and their small contents can feel out of balance.

Maybe this can be improved?

Maybe a single file with all of the utilities that have this same purpose?

**Comment by ybiquitous** [comment]
Thanks for the proposal. Like [`lib/utils/validateTypes.mjs`](https://github.com/stylelint/stylelint/blob/42cd4757369706178fa2600dcca054b068682bd9/lib/utils/validateTypes.mjs), if assembling similar functions makes sense, we should do that. 👍🏼 

Also, such a refactoring should have almost no risks.

**Comment by Mouvedia** [comment]
> if assembling similar functions makes sense, we should do that

👍

**Comment by Mouvedia** [comment]
It seems we are all in favor of having one file that group all of these related utilities; hence I have labeled this issue as ready to implement.
You can bikeshed the file's name in the upcoming PR.
